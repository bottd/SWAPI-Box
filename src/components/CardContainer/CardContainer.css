.CardContainer {
  background-color: #fdde09;
  border-radius: 40px;
  display: flex;
  flex-wrap: wrap;
  height: 60vh;
  justify-content: space-evenly;
  margin: 0 auto;
  opacity: 0.8;
  overflow: scroll;
  padding: 10px;
  width: 90%;
}

.center {
  flex-direction: column;
  align-items: center;
}

.emptyContainerMessage {
  font-family: starWars;
  margin: 0 10px;
  text-align: center;
}

.characterPic {
  height: 30vh;
  mix-blend-mode: multiply;
}

@media screen and (max-width: 1000px) {
  .CardContainer {
    height: 72vh;
  }
}

@media screen and (max-width: 500px) {
  .characterPic {
    height: 200px;
  }
}
