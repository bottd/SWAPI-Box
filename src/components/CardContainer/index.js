import React, { PureComponent } from "react";
import Card from "../Card";
import PropTypes from "prop-types";
import obiWan from "../../images/obi-wan.jpg";
import yoda from "../../images/yoda.png";
import "./CardContainer.css";

class CardContainer extends PureComponent {
  render() {
    const { cardData, handleCardClick, selectedCategory } = this.props;
    let displayText, characterPic;
    
    if (selectedCategory) {
      displayText = "These are not the cards you are looking for!";
      characterPic = obiWan;
    // } 
    // else if (selectedCategory !== 'favorites' && cardData === []) {
    //   displayText = "$ loading $";

    } else {
      displayText = "$ select a category, you must $";
      characterPic = yoda;
    }
    const cards = cardData.map((card, index) => {
      return (
        <Card
          key={Math.random() + index}
          handleCardClick={handleCardClick}
          cardData={card}
        />
      );
    });

    if (cards.length) {
      return <div className="CardContainer">{cards}</div>;
    } else {
      return (
        <div className="CardContainer center">
          <h1 className="emptyContainerMessage">{displayText}</h1>
          <img className="characterPic" alt="obi-wan/yoda" src={characterPic} />
        </div>
      );
    }
  }
}

CardContainer.propTypes = {
  handleCardClick: PropTypes.func.isRequired,
  cardData: PropTypes.array.isRequired,
  selectedCategory: PropTypes.string.isRequired
};

export default CardContainer;
