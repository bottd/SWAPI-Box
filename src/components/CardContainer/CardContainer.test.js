import React from "react";
import { shallow, mount } from "enzyme";
import CardContainer from "./index";

describe("CardContainer", () => {
  it("Renders like snapshot", () => {
    const wrapper = shallow(
      <CardContainer
        cardData={[]}
        handleCardClick={jest.fn()}
        selectedCategory={"planets"}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("Should render message stating no category has been selected", () => {
    const wrapper = shallow(
    <CardContainer
      cardData={[]}
      handleCardClick={jest.fn()}
      selectedCategory={"initial"}
    />
  );
    const message = wrapper.find(".emptyContainerMessage");
    expect(message.text()).toBe("$ select a category, you must $");
  });
  it("Should render message stating there are no given favorites", () => {
    const wrapper = shallow(
    <CardContainer
      cardData={[]}
      handleCardClick={jest.fn()}
      selectedCategory={"planets"}
    />
  );
    const message = wrapper.find(".emptyContainerMessage");
    expect(message.text()).toBe("These are not the cards you are looking for!");
  });

  it("Should not render a message when cards are given", () => {
    const wrapper = shallow(
    <CardContainer
      cardData={[{
        name: 'Luke Skywalker',
        species: 'Human',
        language: 'Galactic Basic',
        population: 200000}]}
      handleCardClick={jest.fn()}
      selectedCategory={"planets"}
    />
    );
    const message = wrapper.find(".emptyContainerMessage");
    expect(message.length).toBe(0);
  });

  it("Should render cards when data is given", () => {
    const wrapper = shallow(
    <CardContainer
      cardData={[{
        name: 'Luke Skywalker',
        species: 'Human',
        language: 'Galactic Basic',
        population: 200000}]}
      handleCardClick={jest.fn()}
      selectedCategory={"planets"}
    />
    );
    const message = wrapper.find("Card");
    expect(message.length).toBe(1);
  });
});
