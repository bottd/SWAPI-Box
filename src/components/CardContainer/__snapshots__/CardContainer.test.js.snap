// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer Renders like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cardData={Array []}
    handleCardClick={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="emptyContainerMessage"
        >
          These are not the cards you are looking for!
        </h1>,
        <img
          className="obiWanPic"
          src="obi-wan.jpg"
        />,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "These are not the cards you are looking for!",
          "className": "emptyContainerMessage",
        },
        "ref": null,
        "rendered": "These are not the cards you are looking for!",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "obiWanPic",
          "src": "obi-wan.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="emptyContainerMessage"
          >
            These are not the cards you are looking for!
          </h1>,
          <img
            className="obiWanPic"
            src="obi-wan.jpg"
          />,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "These are not the cards you are looking for!",
            "className": "emptyContainerMessage",
          },
          "ref": null,
          "rendered": "These are not the cards you are looking for!",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "obiWanPic",
            "src": "obi-wan.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
